Specs/Scene/Cesium3DTileBatchTableResourcesSpec.js

```diff
@@ -0,0 +1,459 @@
+/*global defineSuite*/
+defineSuite([
+        'Scene/Cesium3DTileBatchTableResources',
+        'Core/Cartesian3',
+        'Core/Color',
+        'Core/defaultValue',
+        'Core/HeadingPitchRange',
+        'Core/loadArrayBuffer',
+        'Renderer/ContextLimits',
+        'Scene/Cesium3DTileContentState',
+        'Scene/Cesium3DTileset',
+        'Specs/createScene',
+        'Specs/pollToPromise'
+    ], function(
+        Cesium3DTileBatchTableResources,
+        Cartesian3,
+        Color,
+        defaultValue,
+        HeadingPitchRange,
+        loadArrayBuffer,
+        ContextLimits,
+        Cesium3DTileContentState,
+        Cesium3DTileset,
+        createScene,
+        pollToPromise) {
+    "use strict";
+
+    var scene;
+    var centerLongitude = -1.31995;
+    var centerLatitude = 0.69871;
+
+    var withBatchTableUrl = './Data/Cesium3DTiles/Batched/BatchedWithBatchTable/';
+    var withoutBatchTableUrl = './Data/Cesium3DTiles/Batched/BatchedWithoutBatchTable/';
+    var batchLengthZeroUrl = './Data/Cesium3DTiles/Batched/BatchedNoBuildings/';
+
+    var result = new Color();
+
+    var fakeContentProvider = {
+        getModel : function(index) {
+            return {};
+        }
+    };
+
+    beforeAll(function() {
+        scene = createScene();
+
+        // One building in each data set is always located in the center, so point the camera there
+        var center = Cartesian3.fromRadians(centerLongitude, centerLatitude, 5.0);
+        scene.camera.lookAt(center, new HeadingPitchRange(0.0, -1.57, 10.0));
+    });
+
+    afterAll(function() {
+        scene.destroyForSpecs();
+    });
+
+    afterEach(function() {
+        scene.primitives.removeAll();
+    });
+
+    function verifyRender(tileset) {
```

Comment:
Do we have enough use cases at this point so we are ready to make some common functions in a separate PR?